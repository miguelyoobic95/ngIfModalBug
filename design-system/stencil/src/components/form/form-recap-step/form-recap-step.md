---
name: FormRecapStep
category: Core Components
---

```form-recap-step.html
    <yoo-form-recap-step></yoo-form-recap-step>
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`text`|string|   |the text inside the label|

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|form-recap-step accent|
|Dark|`dark`|form-recap-step dark|
|Success|`success`|form-recap-step success|
|Danger|`danger`|form-recap-step danger|
|Info|`info`|form-recap-step info|
|Warning|`warning`|form-recap-step warning|
|Gradient Accent|`gradient-accent`|form-recap-step gradient-accent|
|Gradient Dark|`gradient-dark`|form-recap-step gradient-grey|
|Gradient Success|`gradient-success`|form-recap-step gradient-success|
|Gradient Danger|`gradient-danger`|form-recap-step gradient-danger|
|Gradient Info|`gradient-info`|form-recap-step gradient-info|
|Gradient Warning|`gradient-warning`|form-recap-step gradient-warning|

## Events
|Attr|Description|
|---|---|---|
|`yooXXXX`|small label|