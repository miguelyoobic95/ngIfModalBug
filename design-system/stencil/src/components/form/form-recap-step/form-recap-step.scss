:host {
    .container {
        margin: 15px;
        padding: 20px;
        border-radius: 6px;
        background-color: cssvar(light);
        box-shadow: $tile-shadow;
        text-align: left;
        &.valid {
            background: cssvar(gradient-success);
            color: cssvar(light);
            .step,
            .title,
            .subtitle {
                color: cssvar(light);
            }
        }
        &.invalid {
            background: cssvar(gradient-danger);
            color: cssvar(light);
            .step,
            .title,
            .subtitle {
                color: cssvar(light);
            }
        }
        .step {
            font-size: 9px;
            color: cssvar(success);
            line-height: 1.31;
            letter-spacing: 1px;
        }
        .title {
            font-size: 15px;
            font-weight: bold;
        }
        .subtitle {
            font-size: 11px;
            color: cssvar(text-color);
        }
    }
}