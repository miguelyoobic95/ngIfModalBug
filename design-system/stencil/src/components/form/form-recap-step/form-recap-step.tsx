import { Component, Element, Prop } from '@stencil/core';
import { ITranslateService } from '@shared/interfaces';

@Component({
    tag: 'yoo-form-recap-step',
    styleUrl: 'form-recap-step.scss',
    scoped: true
})
export class YooFormRecapStepComponent {

    @Prop() stepNumber: number;
    @Prop() mainTitle: string;
    @Prop() subTitle: string;
    @Prop() validity: boolean;

     @Element() host: HTMLStencilElement;

    private translate: ITranslateService = (window as any).translateService;

    render(): JSX.Element {
        return (
            <div class={'container ' + (this.validity === true ? 'valid' : (this.validity === false ? 'invalid' : ''))}>
                <div class="step">{this.translate.get('STEP') + ' ' + this.stepNumber}</div>
                <div class="title">{this.translate.get(this.mainTitle)}</div>
                <div class="subtitle">{this.translate.get(this.subTitle)}</div>
            </div>
        );
    }
}
