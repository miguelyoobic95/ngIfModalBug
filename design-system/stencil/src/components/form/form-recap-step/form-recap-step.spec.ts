import { TestWindow } from '@stencil/core/testing';
import { YooFormRecapStepComponent } from './form-recap-step';

describe('FormRecapStepComponent', () => {
    it('should build', () => {
        expect(new YooFormRecapStepComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach( async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooFormRecapStepComponent],
                html: '<yoo-form-recap-step></yoo-form-recap-step>'
            });
        });

        it ('Should render', async () => {
            expect(element).toMatchSnapshot();
        });
    });

    describe('Events', () => {

        let emitEvent, window;
        beforeEach( async () => {
            window = new TestWindow();
            emitEvent = jest.fn();
        });
    });
});