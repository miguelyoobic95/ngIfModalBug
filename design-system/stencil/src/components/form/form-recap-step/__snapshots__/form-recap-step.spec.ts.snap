// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRecapStepComponent Rendering Should render 1`] = `null`;
